File: PAPI/irdr/src/cirsi/defringe.c
/* defringe.c -- subtract background image to defringe */

#include <stdio.h>
#include <stdlib.h>
#include "irdr.h"

int main(int argc, char *argv[])
{
    int i, nx, ny;
    float *img, *bkg, mode, sigma;

    if (argc != 3)
        eprintf("Usage: %s img.fits bkg.fits\n", argv[0]);

    img = readfits(argv[1], &nx, &ny, NULL, NULL);
    bkg = readfits(argv[2], &nx, &ny, &mode, &sigma);

    for (i = 0; i < nx * ny; i++)
        img[i] += (mode - bkg[i]);          /* subtract out bkg structure */

    writefits(argv[1], argv[1], (char*)img, -32, nx, ny);

    return 0;
}

File: irdr/scripts/cirsi/defringe.pl
#!/usr/bin/env perl
#
# use SExtractor to produce a background map for a coadded dither set, then
# subtract that bkgmap to defringe the image.
#
# Jan 2001
#

die "Usage: defringe.pl irx.r*c?.fits\n"
    unless ($#ARGV >= 0);

$basedir = $ENV{'IRDR_BASEDIR'};

die "Please setenv IRDR_BASEDIR /path/irdr\n"
    unless ($basedir);

$prog = "$basedir/bin/defringe";

die "Expected executable: $prog\n"
    unless (-x $prog);

while ($fn = shift(@ARGV)) {
    $wfn = $fn;
    $wfn =~ s/\.fits/\.weight\.fits/;

    die "Sorry, expected to find $wfn and $fn\n"
        unless (-e $wfn and -e $fn);

    $cmd = "sex $fn -c $basedir/src/config/default.sex -WEIGHT_IMAGE $wfn " .
           " -WEIGHT_TYPE MAP_WEIGHT -CHECKIMAGE_TYPE BACKGROUND " .
           " -CHECKIMAGE_NAME $fn.bkg -FITS_UNSIGNED Y";

    die "Sorry, SExtractor failed on file: $fn\n"
        unless(system($cmd) == 0);

    $cmd = "$prog $fn $fn.bkg";

    print "Running defringing command:\n$cmd\n";

    die "Sorry, command $cmd failed\n"
        unless (system($cmd) == 0);
}
